### PR TITLE
fix: approved account ids

### PR DESCRIPTION
**Pull request by hanakannzashi**
According to the [comments](https://github.com/near/near-sdk-rs/blob/master/near-contract-standards/src/non_fungible_token/core/resolver.rs#L29) and [`nft_token` implementation](https://github.com/near/near-sdk-rs/blob/master/near-contract-standards/src/non_fungible_token/core/core_impl.rs#L412:L415), I think 'Extension Not Exists' vs 'Extension Exist But Value Not Exists' is different. i.e. If a contract doesn't have Approval Extension,  when we get `approved_account_ids`, it should be `None`, but if a contract does have Approval Extension, then we get `approved_account_ids`, it should be at least empty map, NEVER be `None`.

### CODE REVIEW

**Review by frol — APPROVED** [review]
@hanakannzashi This is definitely a good usability improvement!

**Review by miraclx — COMMENTED** [review]
Seems the benefit here (in the case that approval extensions are supported, but the map is empty) is that the error moves from being "Unauthorized" to "Sender not approved" and if approval extensions are not supported, the panic message is "Approval extension is disabled"

Just clearing that out of the way because even though account approvals are supported, but an empty map is provided, WE WILL STILL panic, just with an arguably, better error message.

The only case that only ever successfully executes, is if account approvals are supported (AND the non-empty associative map contains the `sender_id`)

**Comment by frol** [comment]
@miraclx Yes, that is it. It is a minor yet welcome UX improvement. I am not a codeowner here, can you approve it if it looks fine to you?

**Review by miraclx — APPROVED** [review]
Lgtm.

**Pull request merged**
